std/math/emulated/field_binary.go

```diff
@@ -34,6 +33,24 @@ func (f *Field[T]) ToBits(a *Element[T]) []frontend.Variable {
 	return fullBits
 }
 
+// ToBitsCanonical represents the unique bit representation in the canonical
+// format (less that the modulus).
+func (f *Field[T]) ToBitsCanonical(a *Element[T]) []frontend.Variable {
+	// TODO: implement a inline version of this function. We perform binary
+	// decomposition both in the `ReduceStrict` and `ToBits` methods, but we can
+	// essentially do them at the same time.
+	//
+	// If we do this, then also check in places where we use `Reduce` and
+	// `ToBits` after that manually (e.g. in point and scalar marshaling) and
+	// replace them with this method.
+
+	var fp T
+	nbBits := fp.Modulus().BitLen()
+	ca := f.ReduceStrict(a)
+	bts := f.ToBits(ca)
```

Comment:
don't we have an option in `ToBits` to give the number of bits we want and reduce the number of constraints generated by `ToBits`?

Reply:
Currently for `Field.ToBits` we don't and I think that it would be a good addition.

But I think I would do it in a separate issue/PR to additionally also handle the TODO in this method. Imo in places we currently use the `ToBits` method we use all bits.